### PR TITLE
Fixed edge cases when loading typing.Dict, typing.List and typing.ClassVar

### LINKED ISSUE #73: `installation.load()` does not handle dict correctly
simple example
```py
from databricks.sdk.service.compute import ClusterDetails

installation = MockInstallation(...)
installation.load(ClusterDetails, filename=...)
```
fails with 
```py
databricks.labs.blueprint.installation.SerdeError: spark_conf: unknown: <class 'dict'>
```

### PR DESCRIPTION
**Pull request by nkvuong**
Resolves #73

### CODE REVIEW

File: src/databricks/labs/blueprint/installation.py [hunk 1]
```diff
@@ -623,6 +624,8 @@ def _unmarshal(cls, inst: Any, path: list[str], type_ref: type[T]) -> T | None:
         if isinstance(type_ref, _GenericAlias):
             if not inst:
                 return None
+            if type_ref.__origin__ in (dict, list):
+                return cls._unmarshal_generic(inst, path, type_ref)
```
Comment (nfx) [inline src/databricks/labs/blueprint/installation.py:628]: Could you add a test?

File: src/databricks/labs/blueprint/installation.py [hunk 2]
```diff
@@ -651,6 +654,9 @@ def _unmarshal_dataclass(cls, inst, path, type_ref):
         from_dict = {}
         fields = getattr(type_ref, "__dataclass_fields__")
         for field_name, hint in get_type_hints(type_ref).items():
+            origin = getattr(hint, "__origin__", None)
+            if origin is typing.ClassVar:
```
Comment (nfx) [inline src/databricks/labs/blueprint/installation.py:658]: And for this?

**Review by nfx — APPROVED** [review]

File: src/databricks/labs/blueprint/installation.py (hunk 1 above)
Comment (nkvuong) [reply src/databricks/labs/blueprint/installation.py:629]: done

File: src/databricks/labs/blueprint/installation.py (hunk 2 above)
Comment (nkvuong) [reply src/databricks/labs/blueprint/installation.py:659]: done

**Comment by nkvuong** [comment]
combined some branches, to avoid `Too many return statements (12/11) (too-many-return-statements)`

**Review by nfx — APPROVED** [review]
Thank you 🙏🏻

**Pull request merged**